File: 剑指Offer/从上到下打印二叉树/printBST.java
import java.util.ArrayList;
import java.util.Queue;
import java.util.LinkedList;
/**
public class TreeNode {
    int val = 0;
    TreeNode left = null;
    TreeNode right = null;
 
    public TreeNode(int val) {
        this.val = val;
 
    }
 
}
*/
public class Solution {
    public ArrayList<Integer> PrintFromTopToBottom(TreeNode root) {
        Queue<TreeNode> que = new LinkedList<TreeNode>();
        ArrayList<Integer> res = new ArrayList<Integer>();
        if(root == null){
            return res;
        }
        que.offer(root);
        while(que.size()>0){
            TreeNode node = que.poll();
            if(node.left!=null){
                que.offer(node.left);
            }
            if(node.right!=null){
                que.offer(node.right);
            }
            res.add(node.val);
        }
        return res;
    }
}

import java.util.ArrayList;
/**
public class TreeNode {
    int val = 0;
    TreeNode left = null;
    TreeNode right = null;
 
    public TreeNode(int val) {
        this.val = val;
 
    }
 
}
*/
public class Solution {
    public ArrayList<Integer> PrintFromTopToBottom(TreeNode root) {
        ArrayList<TreeNode> que = new ArrayList<TreeNode>();
        ArrayList<Integer> res = new ArrayList<Integer>();
        if(root == null){
            return res;
        }
        que.add(root);
        while(que.size()>0){
            TreeNode node = que.remove(0);
            if(node.left!=null){
                que.add(node.left);
            }
            if(node.right!=null){
                que.add(node.right);
            }
            res.add(node.val);
        }
        return res;
    }
}

File: 剑指Offer/从上到下打印二叉树/printBST.py
# -*- coding:utf-8 -*-
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
class Solution:
    # 返回从上到下每个节点值列表，例：[1,2,3]
    def PrintFromTopToBottom(self, root):
        # write code here
        res = []
        if not root:
            return res
        sta = [root]
        while len(sta)>0:
            node = sta.pop(0)
            if node.left:
                sta.append(node.left)
            if node.right:
                sta.append(node.right)
            res.append(node.val)
        return res